buildSrc/src/main/kotlin/Semver.kt

```diff
@@ -0,0 +1,26 @@
+/**
+ * Get the versioncode for a given semver. Returns null if value is invalid.
+ *
+ * Sample output:
+ * 0.0.0 -> 0
+ * 1.1.1 -> 10101
+ * 0.7.0 -> 700
+ * 99.99.99 -> 999999
+ */
+fun getVersionCode(semver: String): Int? {
+	val parts = semver.splitToSequence('.')
+		.take(3)
+		.map { it.toIntOrNull() }
+		.filterNotNull()
```

Comment:
I think you could use `mapNotNull` here.

Reply:
Yup, IDE didn't hint it and I didn't think about it when writing the function, changed it now.